### ISSUE TITLE
search_users - allow search by `query`

### ISSUE THREAD

**Issue by robeden** [pycontribs/jira#924]
**Is your feature request related to a problem? Please describe.**
The `search_users` function currently only allows searching by `username` which is not supported when in strict GDPR mode. In that case, this error is returned:
> The query parameter 'username' is not supported in GDPR strict mode.

Currently, I don't believe there is a way around this using public functions in the library.

**Describe the solution you'd like**
One option would be to add a parameter such as the following to the function:
`useQuery = False`. When done, then `username` key in the params would be replaced with `query`. I think that's all that needs to be done.

**Describe alternatives you've considered**
I can get around this problem by using the private `_fetch_pages` function:
```
found_users = _jira._fetch_pages(User, None, 'user/search', 0, 50, {"query": "me@example.invalid"})
```
... but obviously this is gross.

**Comment by svermeulen** [comment]
The comment suggests that the given 'username' parameter can be an email or name too.  Maybe that value should just be replaced by query?  This is what I'm doing [here](https://github.com/svermeulen/jira/commit/24ae0922387f8a182b7e36584d393450410655ca)

**Comment by robeden** [comment]
That would work. The only criticism would be that it’s a behavior change and could be surprising. The parameter with a default value would keep the API and behavior unchanged, but allow for using the new behavior if desired.

**Comment by vikoalucard** [comment]
Hi, I get the same problem, but can't even make the workaround to work.

> I can get around this problem by using the private `_fetch_pages` function:
> ```
> found_users = _jira._fetch_pages(User, None, 'user/search', 0, 50, {"query": "me@example.invalid"})
> ```

On my side:
`allusers = jiratarget._fetch_pages(User, None, 'user/search', 0, 50, {"query": "username=noreplay@example.com"})`
=> gets this error:
`response text = {"errorMessages":["The username query parameter was not provided"],"errors":{}}`

**Comment by adehad** [comment]
Following #927 we now do allow a `query` argument to be passed to `search_users()`

**Comment by mattdough** [comment]
Should this be in the 3.0.1 version of the package? 

I've currently got that installed when looking at the search_users definition it doesn't have the query added.

**Comment by adehad** [comment]
@mattdough this is included in the pre-release: `pip install jira==3.1.0rc1`

I can see it in the documentation here https://jira.readthedocs.io/api.html#jira.client.JIRA.search_users:
![image](https://user-images.githubusercontent.com/26027314/141084629-64731a0f-bd2f-4280-9ad6-f50d752b4d4a.png)
